### PR TITLE
Allow Logary to handle exceptions logged via the Facade

### LINKED ISSUE #168: Logging an exception from Facade crashes Logary (proper)
I've created a [sample that crashes Logary (proper)](https://github.com/adamchester/logary/commit/9e42bd782fa8dce202d54eba81b5f37d636c344e).

The stacktrace follows:
```
Unhandled Exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: Method may only be called on a Type for which Type.IsGenericParameter is true.
   at System.RuntimeType.get_DeclaringMethod()
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.RuntimePropertyInfo.GetValue(Object obj, Object[] index)
   at Logary.Internals.Map.ofObject@251-5.Invoke(PropertyInfo pi) in C:\My\repo\logary-2\src\Logary\Internals.fs:line 251
   at Logary.Internals.Map.ofObject@204.Invoke(Object _arg2) in C:\My\repo\logary-2\src\Logary\Internals.fs:line 251
   at Logary.ValueModule.OfObject(Object _arg1) in C:\My\repo\logary-2\src\Logary\DataModel.fs:line 248
   at Logary.ValueModule.OfObject@250-4.Invoke(String _arg1, Object v) in C:\My\repo\logary-2\src\Logary\DataModel.fs:line 250
   at Microsoft.FSharp.Collections.MapTreeModule.mapiOpt[a,b,c](FSharpFunc`3 f, MapTree`2 m)
   at Microsoft.FSharp.Collections.MapTreeModule.mapiOpt[a,b,c](FSharpFunc`3 f, MapTree`2 m)
   at Microsoft.FSharp.Collections.FSharpMap`2.Map[b](FSharpFunc`2 f)
   at Logary.ValueModule.OfObject(Object _arg1) in C:\My\repo\logary-2\src\Logary\DataModel.fs:line 248
   at Logary.ValueModule.OfObject@238-2.Invoke(Object _arg1) in C:\My\repo\logary-2\src\Logary\DataModel.fs:line 238
   at Microsoft.FSharp.Collections.IEnumerator.map@111.DoMoveNext(b& )
   at Microsoft.FSharp.Collections.IEnumerator.MapEnumerator`1.System-Collections-IEnumerator-MoveNext()
   at Microsoft.FSharp.Collections.SeqModule.ToList[T](IEnumerable`1 source)
   at Logary.ValueModule.OfObject(Object _arg1) in C:\My\repo\logary-2\src\Logary\DataModel.fs:line 236
   at Logary.ValueModule.OfObject@250-4.Invoke(String _arg1, Object v) in C:\My\repo\logary-2\src\Logary\DataModel.fs:line 250
   at Microsoft.FSharp.Collections.MapTreeModule.mapiOpt[a,b,c](FSharpFunc`3 f, MapTree`2 m)
   at Microsoft.FSharp.Collections.MapTreeModule.mapiOpt[a,b,c](FSharpFunc`3 f, MapTree`2 m)
   at Microsoft.FSharp.Collections.MapTreeModule.mapiOpt[a,b,c](FSharpFunc`3 f, MapTree`2 m)
   at Microsoft.FSharp.Collections.MapTreeModule.mapiOpt[a,b,c](FSharpFunc`3 f, MapTree`2 m)
   at Microsoft.FSharp.Collections.MapTreeModule.mapiOpt[a,b,c](FSharpFunc`3 f, MapTree`2 m)
   at Microsoft.FSharp.Collections.FSharpMap`2.Map[b](FSharpFunc`2 f)
   at Logary.ValueModule.OfObject(Object _arg1) in C:\My\repo\logary-2\src\Logary\DataModel.fs:line 248
   at Logary.ValueModule.OfObject@250-4.Invoke(String _arg1, Object v) in C:\My\repo\logary-2\src\Logary\DataModel.fs:line 250
   at Microsoft.FSharp.Collections.MapTreeModule.mapiOpt[a,b,c](FSharpFunc`3 f, MapTree`2 m)
   at Microsoft.FSharp.Collections.MapTreeModule.mapiOpt[a,b,c](FSharpFunc`3 f, MapTree`2 m)
   at Microsoft.FSharp.Collections.MapTreeModule.mapiOpt[a,b,c](FSharpFunc`3 f, MapTree`2 m)
   at Microsoft.FSharp.Collections.FSharpMap`2.Map[b](FSharpFunc`2 f)
   at Logary.ValueModule.OfObject(Object _arg1) in C:\My\repo\logary-2\src\Logary\DataModel.fs:line 248
   at Logary.ValueModule.OfObject@238-2.Invoke(Object _arg1) in C:\My\repo\logary-2\src\Logary\DataModel.fs:line 238
   at Microsoft.FSharp.Collections.IEnumerator.map@111.DoMoveNext(b& )
   at Microsoft.FSharp.Collections.IEnumerator.MapEnumerator`1.System-Collections-IEnumerator-MoveNext()
   at Microsoft.FSharp.Collections.SeqModule.ToList[T](IEnumerable`1 source)
   at Logary.ValueModule.OfObject(Object _arg1) in C:\My\repo\logary-2\src\Logary\DataModel.fs:line 236
   at Logary.MessageModule.SetFieldsFromMap@1146.Invoke(KeyValuePair`2 _arg1) in C:\My\repo\logary-2\src\Logary\DataModel.fs:line 1146
   at Microsoft.FSharp.Collections.IEnumerator.map@111.DoMoveNext(b& )
   at Microsoft.FSharp.Collections.IEnumerator.MapEnumerator`1.System-Collections-IEnumerator-MoveNext()
   at Microsoft.FSharp.Collections.SeqModule.ToList[T](IEnumerable`1 source)
   at Logary.MessageModule.SetFieldsFromMap(FSharpMap`2 m, Message msg) in C:\My\repo\logary-2\src\Logary\DataModel.fs:line 1145
   at Logary.Adapters.Facade.LoggerAdapter.toMsg(String[] fallbackName, Object o) in C:\My\repo\logary-2\src\adapters\Logary.Adapters.Facade\Logary.Adapters.Facade.fs:line 59
   at Logary.Adapters.Facade.LoggerAdapter.|Log|LogWithAck|LogSimple|(IInvocation _arg1_0, String[] _arg1_1) in C:\My\repo\logary-2\src\adapters\Logary.Adapters.Facade\Logary.Adapters.Facade.fs:line 90
   at Logary.Adapters.Facade.LoggerAdapter.I.Castle-DynamicProxy-IInterceptor-Intercept(IInvocation invocation) in C:\My\repo\logary-2\src\adapters\Logary.Adapters.Facade\Logary.Adapters.Facade.fs:line 143
   at Castle.DynamicProxy.AbstractInvocation.Proceed()
   at Libryy.Core.simpleWorkThatCatchesAndLogsAnErrorAndException(Logger logger) in C:\My\repo\logary-2\examples\Libryy\Libryy.fs:line 43
   at Servizz.Program.main(String[] argv) in C:\My\repo\logary-2\examples\Servizz\Program.fs:line 34
W 2016-08-01T10:38:03.0290450+00:00: Hey haf! [Libryy.Core.work]
  user => "haf"
D 2016-09-25T03:19:38.7369499+00:00: Got 42 from Libryy [Libryy]
  workResult => 42
Press any key to continue . . .
```

### PR DESCRIPTION
**Pull request by adamchester**
This fixes #168

### CODE REVIEW

**Comment by adamchester** [comment]
Any thoughts on this?

**Comment by haf** [comment]
Yes, I tried the new review feature and didn't realise that `Pending` meant that you couldn't see it.

File: src/Logary/DataModel.fs [hunk 1]
```diff
@@ -183,6 +183,29 @@ type Value =
...
+        if e.Source <> null then
+          yield "source", String e.Source
+        if e.HelpLink <> null then
+          yield "helpLink", String e.HelpLink
+        if e.HResult <> 0 then
+          yield "hResult", Int64 (int64 e.HResult)
+        if e.Data <> null && e.Data.Count > 0 then
+          yield "data", valueOfObject e.Data ]
+
+    Map.ofSeq <|
```
Comment (adamchester) [inline src/Logary/DataModel.fs]: I only moved this block of code as-is. Also, I'm almost thinking it does bring some benefit here. Thoughts?

Comment (haf) [inline src/Logary/DataModel.fs]: Yes, agreed. Let's keep it.

**Comment by haf** [comment]
LGTM! :)

**Pull request merged**